pkg/quantile/ddsketch.go

```diff
@@ -0,0 +1,217 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package quantile
+
+import (
+	"fmt"
+	"math"
+	"sort"
+
+	"github.com/DataDog/datadog-agent/pkg/quantile/summary"
+	"github.com/DataDog/sketches-go/ddsketch"
+	"github.com/DataDog/sketches-go/ddsketch/mapping"
+	"github.com/DataDog/sketches-go/ddsketch/store"
+)
+
+const (
+	maxIndex = math.MaxInt16
+)
+
+// createDDSketchWithSketchMapping takes a DDSketch and returns a new DDSketch
+// with a logarithmic mapping that matches the Sketch parameters.
+func createDDSketchWithSketchMapping(c *Config, inputSketch *ddsketch.DDSketch) (*ddsketch.DDSketch, error) {
+	// Create positive store for the new DDSketch
+	positiveStore := store.NewDenseStore()
+
+	// Create negative store for the new DDSketch
+	negativeStore := store.NewDenseStore()
+
+	// Take parameters that match the Sketch mapping, and create a LogarithmicMapping out of them
+	gamma := c.gamma.v
+	offset := float64(c.norm.bias) + 0.5
```

Comment:
Nit.: for future reference, maybe add a comment to explain the 0.5 shift (taking floor value in DDSketch vs. rounding to nearest integer in the agent sketch).